net/sixlowpan/sixlowpan_tcpsend.c

```diff
@@ -660,6 +660,11 @@ static int sixlowpan_send_packet(FAR struct socket *psock,
               ret = net_timedwait(&sinfo.s_waitsem, timeout);
               if (ret != -ETIMEDOUT || acked == sinfo.s_acked)
```

Comment:
Condition `ret != -ETIMEDOUT` here and then checking for `if (ret == -ETIMEOUT)` makes me feel a bit uncomfortable, but I assume that we are seeking for the `acked == sinfo.s_acked` condition here.